### PR TITLE
Passing props into react-autosuggest

### PR DESCRIPTION
**Pull request by AskseL**
Adds ability for passing props into React-Autosuggest using AutocompleteInput (such as focusInputOnSuggestionClick)

### CODE REVIEW

File: packages/ra-ui-materialui/src/input/AutocompleteInput.js [hunk 1]
```diff
@@ -444,6 +444,7 @@ export class AutocompleteInput extends React.Component {
             source,
             className,
             options,
+            autosuggestProps = {}
```
Comment (djhi) [inline packages/ra-ui-materialui/src/input/AutocompleteInput.js]: We should use the same convention as for the other input/field components: pass all unknown props to the root component, here `Autosuggest`. 

I suggest replacing this prop by `...rest`

**Comment by fzaninotto** [comment]
I've rebased the `next` branch on `master` (it was lagging behind the latest minor). Could you please rebase your PR on the new `next`?

**Comment by AskseL** [comment]
changed custom props to {...rest}

File: packages/ra-ui-materialui/src/input/AutocompleteInput.js (hunk 1 above)
Comment (AskseL) [reply packages/ra-ui-materialui/src/input/AutocompleteInput.js]: done

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**